### PR TITLE
Added Root Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this meta ticket https://github.com/ignitionrobotics/sdformat/issues/931

Require
 - https://github.com/ignitionrobotics/sdformat/pull/950
 - https://github.com/ignitionrobotics/sdformat/pull/952
 - https://github.com/ignitionrobotics/sdformat/pull/961
 - https://github.com/ignitionrobotics/sdformat/pull/963
 - https://github.com/ignitionrobotics/sdformat/pull/964

## Summary

Added Root Python interface

## Test it

```python
from sdformat import Root

root = Root()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pyRoot.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    .def("set_model", &sdf::Root::SetModel,
+         pybind11::return_value_policy::reference_internal,
+         "Set the model object. This will override any existing model, "
+         "actor, and light object.")
+    .def("add_world", &sdf::Root::AddWorld,
+         "Add a world to the root.")
+    .def("clear_worlds", &sdf::Root::ClearWorlds,
+         "Remove all worlds.")
+    .def("update_graphs", &sdf::Root::UpdateGraphs,
+         "ecreate the frame and pose graphs for the worlds and model "
```
Comment (scpeters) [inline python/src/sdf/pyRoot.cc:85]: nit: `ecreate` -> `Recreate`

File: python/test/pyRoot_TEST.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from ignition.math import Vector3d, Pose3d
+from sdformat import Error, Root, SDF_VERSION, SDF_PROTOCOL_VERSION, World
```
Comment (scpeters) [inline python/test/pyRoot_TEST.py:17]: I believe `SDF_VERSION` and `SDF_PROTOCOL_VERSION` are not yet defined

Comment (ahcorde) [reply python/test/pyRoot_TEST.py:17]: Done https://github.com/ignitionrobotics/sdformat/pull/967/commits/6d58711b2c3eb125fe7034ae179a359853a36958

File: python/src/sdf/pyRoot.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    .def("world_by_index",
+         pybind11::overload_cast<uint64_t>(
+           &sdf::Root::WorldByIndex),
+         pybind11::return_value_policy::reference_internal,
+         "Get a world based on an index.")
+     .def("world_name_exists", &sdf::Root::WorldNameExists,
+          "Get whether a world name exists.")
+    .def("model", &sdf::Root::Model,
+         pybind11::return_value_policy::reference_internal,
+         "Get a pointer to the model object if it exists.")
```
Comment (scpeters) [inline python/src/sdf/pyRoot.cc:75]: this doesn't return a pointer in python: "Get a model object if it exists"

File: python/test/pyRoot_TEST.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+import unittest
+
+
+class RootTEST(unittest.TestCase):
+
+  def test_default_construction(self):
+    root = Root()
+    # self.assertEqual(SDF_VERSION, root.version())
```
Comment (scpeters) [inline python/test/pyRoot_TEST.py:25]: can this expectation be enabled?

Comment (scpeters) [reply python/test/pyRoot_TEST.py]: enabled in https://github.com/ignitionrobotics/sdformat/pull/967/commits/1f0c329751801eda7c45911dc2fa73d87767dfc2

File: python/src/sdf/pyRoot.cc (hunk 3 above)
Comment (scpeters) [reply python/src/sdf/pyRoot.cc]: edited in https://github.com/ignitionrobotics/sdformat/pull/967/commits/1f0c329751801eda7c45911dc2fa73d87767dfc2

**Review by scpeters — APPROVED** [review]

**Pull request merged**